### PR TITLE
add site configuration page, frontend and backend

### PR DESCRIPTION
**Pull request by bkmaibach**
## Description
site configuration (maintenance mode boolean and text)

## What type of PR is this? (check all applicable)

- [x ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📦 Chore (Release)
- [ ] ✅ Test

## Related Tickets & Documents

https://hous-bssb.atlassian.net/browse/BPHH-801

## Steps to QA

log in as super admin
go to site configuration
set switch and text input
save and refresh
try logged out, different roles etc

### CODE REVIEW

**Comment by karenkun** [comment]
Tested this and if I'm logged in as submitter, reviewer, reviewer_manager I won't see that new maintenance alert until I reload the page. Is that okay??

File: app/controllers/api/site_configuration_controller.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  def index
+    authorize :site_configuration, :index?
+    render_success @site_configuration
+  end
+
+  def create
+    authorize :site_configuration, :create?
+    @site_configuration.update(site_configuration_params)
+    render_success @site_configuration, "site_configuration.update_success"
```
Comment (taw27) [inline app/controllers/api/site_configuration_controller.rb:15]: no need to handle errors?

File: app/models/site_configuration.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+  def self.instance
+    first_or_create
+  end
+
+  private
+
+  # A private method to ensure only one record exists
+  def ensure_single_record
+    throw :abort if SiteConfiguration.count > 0
```
Comment (taw27) [inline app/models/site_configuration.rb:13]: why not add normal validation errors.add ?

File: app/controllers/api/site_configuration_controller.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+class Api::SiteConfigurationController < Api::ApplicationController
+  before_action :set_site_configuration, only: %i[index create]
+
+  skip_after_action :verify_policy_scoped
+  after_action :verify_authorized
```
Comment (taw27) [inline app/controllers/api/site_configuration_controller.rb:5]: isn't this only for index?

**Review by taw27 — CHANGES_REQUESTED** [review]

File: app/controllers/api/site_configuration_controller.rb (hunk 3 above)
Comment (bkmaibach) [reply app/controllers/api/site_configuration_controller.rb:5]: the reason these are here is to override the same in the parent, which specifies actions that dont exist in this subclass and cause errors

**Review by taw27 — APPROVED** [review]

**Pull request merged**